Sources/LanguageServerProtocol/SupportTypes/ServerCapabilities.swift

```diff
@@ -459,6 +477,54 @@ public struct CompletionOptions: WorkDoneProgressOptions, Codable, Hashable {
     self.completionItem = completionItem
     self.workDoneProgress = workDoneProgress
   }
+
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    if case .bool(let value) = dictionary["resolveProvider"] {
+      resolveProvider = value
+    }
+
+    if let arrayAny = dictionary["triggerCharacters"] {
+      triggerCharacters = [String](fromLSPArray: arrayAny)
+    }
+
+    if let arrayAny = dictionary["allCommitCharacters"] {
+      allCommitCharacters = [String](fromLSPArray: arrayAny)
+    }
+
+    if case .dictionary(let dict) = dictionary["completionItem"] {
+      completionItem = CompletionItemOptions(fromLSPDictionary: dict)
+    }
+
+    if case .bool(let value) = dictionary["workDoneProgress"] {
+      workDoneProgress = value
+    }
+  }
+
+  public func encodeToLSPAny() -> LSPAny {
+    var dict: [String: LSPAny] = [:]
+    
+    if let resolveProvider {
+      dict["resolveProvider"] = .bool(resolveProvider)
+    }
+    
+    if let triggerCharacters {
+      dict["triggerCharacters"] = triggerCharacters.encodeToLSPAny()
+    }
+
+    if let allCommitCharacters {
+      dict["allCommitCharacters"] = allCommitCharacters.encodeToLSPAny()
+    }
+
+    if let completionItem {
+      dict["completionItem"] = completionItem.encodeToLSPAny()
+    }
+
+    if let workDoneProgress {
+      dict["workDoneProgress"] = .bool(workDoneProgress)
+    }
```

Comment:
Just for my understanding: We didn’t use to include `completionItem` and `workDoneProgress` in the `LSPAny` encoded value, right? I think including it is a good but want to make sure it’s a deliberate choice.

Reply:
yes, I added it